vendor/github.com/shirou/gopsutil/internal/common/common_linux.go

```diff
@@ -9,15 +9,24 @@ import (
 )
 
 func DoSysctrl(mib string) ([]string, error) {
-	err := os.Setenv("LC_ALL", "C")
-	if err != nil {
-		return []string{}, err
+	hostEnv := os.Environ()
+	foundLC := false
+	for i, line := range hostEnv {
+		if strings.HasPrefix(line, "LC_ALL") {
+			hostEnv[i] = "LC_ALL=C"
+			foundLC = true
+		}
+	}
+	if !foundLC {
+		hostEnv = append(hostEnv, "LC_ALL=C")
 	}
 	sysctl, err := exec.LookPath("/sbin/sysctl")
 	if err != nil {
 		return []string{}, err
 	}
-	out, err := exec.Command(sysctl, "-n", mib).Output()
+	cmd := exec.Command(sysctl, "-n", mib)
+	cmd.Env = hostEnv
```

Comment:
Since you're setting the execution environment for just the `sysctl` call, wouldn't it be easier to just always override with `LC_ALL=C`? The locale change has limited scope and should not affect Nomad executor. In fact, the original code was done to avoid any possible locale-induced parsing problem, so `LC_ALL=C` is a desirable thing.